### PR TITLE
lxd/storage: Implement security.unmapped

### PR DESCRIPTION
**Pull request by stgraber**
Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

### CODE REVIEW

File: doc/api-extensions.md [hunk 1]
```diff
@@ -607,3 +607,12 @@ This adds the following new endpoint (see [RESTful API](rest-api.md) for details
 * `PUT /1.0/storage-pools/<pool>/volumes/<type>/<volume>/snapshots/<name>`
 * `POST /1.0/storage-pools/<pool>/volumes/<type>/<volume>/snapshots/<name>`
 * `DELETE /1.0/storage-pools/<pool>/volumes/<type>/<volume>/snapshots/<name>`
+
+## storage\_unmapped
+Introduces a new `security.unmapped` boolean on storage volumes.
```
Comment (brauner) [inline doc/api-extensions.md]: `security.storage.unmapped` maybe?

Comment (stgraber) [reply doc/api-extensions.md]: Why? it's a property on a storage volume, so can't really be anything but storage :)

Comment (brauner) [reply doc/api-extensions.md]: I thought the scoping might make it easier to tell what that property is for and what it applies to and also makes it easier when we add keys later. :)

Comment (stgraber) [reply doc/api-extensions.md]: Still feels redundant and un-needed when scoped to a custom storage volume config key.

I thought of calling it `security.idmap.unmapped` before to better line up with containers, but the problem with that approach is that `security.unmapped` will still be relevant even when `security.idmap` isn't (shiftfs) as it'll then mean to not use shiftfs keeping the disk unmapped.

**Pull request merged**